pkg/apis/batch/validation/validation.go

```diff
@@ -223,6 +227,9 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 			if spec.MaxFailedIndexes != nil && *spec.MaxFailedIndexes > maxFailedIndexesForIndexedJob {
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("maxFailedIndexes"), *spec.MaxFailedIndexes, fmt.Sprintf("must be less than or equal to %d", maxFailedIndexesForIndexedJob)))
 			}
+			if spec.Completions != nil && spec.SuccessPolicy != nil {
+				allErrs = append(allErrs, validateSuccessPolicy(spec, fldPath.Child("successPolicy"))...)
```

Comment:
How does completions impact successPolicy validations?  I would expect to validateSuccessPolicy whenever successPolicy != nil.

Reply:
I believe that we don't need to verify successPolicy against Jobs with null completions since this JobSuccessPolicy is provided only Indexed Jobs.

Reply:
> I believe that we don't need to verify successPolicy against Jobs with null completions since this JobSuccessPolicy is provided only Indexed Jobs.

Then let's explicitly disallow when `.spec.Completions == nil`

Reply:
> > I believe that we don't need to verify successPolicy against Jobs with null completions since this JobSuccessPolicy is provided only Indexed Jobs.
> 
> Then let's explicitly disallow when `.spec.Completions == nil`

I already have added such a validation in Line253.
So, does it mean that I should move those to here?

Reply:
If `completionMode != nil && *spec.CompletionMode != batch.NonIndexedCompletion && spec.Completions == nil`, where is successPolicy validated?

If you added SuccessPolicy validation outside of any other conditional, this would be very clear.  Nesting like this, the dependency between

Reply:
it sounds reasonable. I'll try to put all validations for successPolicy outside this nest.
Thanks.

Reply:
It looks like this is resolved. If so, please mark it as resolved.

Reply:
Done.

Reply:
> It looks like this is resolved. If so, please mark it as resolved.

Yea, that's right. I addressed this comment right, now. Thanks.